### PR TITLE
Add pagination to get_exchange_ids() function

### PR DESCRIPTION
**Pull request by KendallWeihe**
This adds tbdex pagination for the `get_exchange_ids()` function from-APID-all-the-way-through-the-example app

Reference https://github.com/TBD54566975/tbdex/tree/main/specs/http-api#pagination

### CODE REVIEW

File: crates/tbdex/src/http_client/exchanges.rs
Comment (KendallWeihe) [inline crates/tbdex/src/http_client/exchanges.rs:169]: we can improve this in due time, but for now this is the most efficient way. namely, we may consider using a dedicated `Url` type, but which will require additional considerations (namely it may require adding a new dependency which must pass licensing considerations)

File: examples/hosted-wallet-kt/pfi/build.gradle.kts
Comment (KendallWeihe) [inline examples/hosted-wallet-kt/pfi/build.gradle.kts:33]: I want to change this and the other gradle build file to use the latest snapshot, once that's deployed, whereafter we can merge this PR (assuming ✅)

File: crates/tbdex/src/http_client/exchanges.rs
Comment (KendallWeihe) [reply crates/tbdex/src/http_client/exchanges.rs:169]: btw we need to add pagination to other functions too, so we can improve DRY and the like at such time

File: examples/hosted-wallet-kt/pfi/build.gradle.kts
Comment (KendallWeihe) [reply examples/hosted-wallet-kt/pfi/build.gradle.kts]: this is done

File: bindings/tbdex_uniffi/src/http_client/exchanges.rs
Comment (angiejones) [inline bindings/tbdex_uniffi/src/http_client/exchanges.rs:82]: if the query_param is absent, does it return the full list?

Comment (KendallWeihe) [reply bindings/tbdex_uniffi/src/http_client/exchanges.rs:82]: that would be logic implemented on the PFI-side, but yes if not included then the expectation would be the full set of exchange IDs

perhaps we should add that language to [the spec](https://github.com/TBD54566975/tbdex/tree/main/specs/http-api#pagination), or perhaps we should have a maximum as the default and include pagination on the response (@jiyoontbd @mistermoe)

Comment (jiyoontbd) [reply bindings/tbdex_uniffi/src/http_client/exchanges.rs:82]: if there's a million exchange between Alice and PFI already, it might be just a tad bit difficult for PFI to return all that in one response

so i agree with the idea of having a max. we could have it paginated by default showing the last X exchange... i don't really know what X should be. 50? 100?

Comment (jiyoontbd) [reply bindings/tbdex_uniffi/src/http_client/exchanges.rs:82]: this is just a list of ids tho, right, and not the full list of messages (which is an Exchange)?

File: crates/tbdex/src/http_client/exchanges.rs
Comment (decentralgabe) [inline crates/tbdex/src/http_client/exchanges.rs:171]: Are these params specified by the client or server?

File: docs/API_DESIGN.md
Comment (decentralgabe) [inline docs/API_DESIGN.md:633]: Non blocking but is it worth having a DID type?

**Review by decentralgabe — APPROVED** [review]
I'm no rust expert but LGTM!

File: docs/API_DESIGN.md
Comment (KendallWeihe) [reply docs/API_DESIGN.md:633]: Yeah, it's an idea worthy of consideration. I default to primitive types (strings) where possible, because I like to minimize cross-dependency usage (a DID type would originate, and in fact already exists, inside of `web5-rs`), but that's my bias.

File: crates/tbdex/src/http_client/exchanges.rs
Comment (KendallWeihe) [reply crates/tbdex/src/http_client/exchanges.rs:171]: Client. They're HTTP query params submitted by the client to query a subset of exchange IDs from the PFI.

File: crates/tbdex/src/http_client/exchanges.rs
Comment (KendallWeihe) [reply crates/tbdex/src/http_client/exchanges.rs:165]: Good question. The `get_service_endpoint` is not something exposed internally, so just an internal function for the sake of DRY, and it specifically searched for the service with the type equal to `PFI` which is the only applicable DID service method in the context of the `tbdex::http_client` module.

Comment (decentralgabe) [reply crates/tbdex/src/http_client/exchanges.rs:165]: there could still be multiple 😄

Comment (KendallWeihe) [reply crates/tbdex/src/http_client/exchanges.rs:165]: Very fair point! https://github.com/TBD54566975/tbdex-rs/issues/108#issuecomment-2278305790

**Pull request merged**
